packages/calcite-components/src/components/block-group/block-group.e2e.ts

```diff
@@ -0,0 +1,550 @@
+            newIndex: testWindow.newIndex,
+            fromEl: testWindow.fromEl,
+            toEl: testWindow.toEl,
+            el: testWindow.el,
+          };
+        });
+
+        const id = "component1";
+
+        expect(results.calledTimes).toBe(++totalMoves);
+        expect(results.newIndex).toBe(newIndex);
+        expect(results.oldIndex).toBe(oldIndex);
+        expect(results.fromEl).toBe(id);
+        expect(results.toEl).toBe(id);
+        expect(results.el).toBe("one");
+      }
+
+      await assertReorder("down", ["two", "one", "three"], 1, 0);
+      await assertReorder("down", ["two", "three", "one"], 2, 1);
+      await assertReorder("down", ["two", "three", "one"], 2, 2);
+
+      await assertReorder("up", ["two", "one", "three"], 1, 2);
+      await assertReorder("up", ["one", "two", "three"], 0, 1);
+      await assertReorder("up", ["one", "two", "three"], 0, 0);
+
+      await assertReorder("bottom", ["two", "three", "one"], 2, 0);
+      await assertReorder("top", ["one", "two", "three"], 0, 2);
+    });
+
+    it("moves using a keyboard", async () => {
+      const page = await newE2EPage();
+      const group = "my-group";
+      await page.setContent(
+        html`<calcite-block-group id="component1" group="${group}" drag-enabled>
+            <calcite-block id="one" heading="one" label="One"></calcite-block>
+            <calcite-block id="two" heading="two" label="Two"></calcite-block>
+          </calcite-block-group>
+          <calcite-block-group id="component2" group="${group}" drag-enabled>
+            <calcite-block id="three" heading="three" label="Three"></calcite-block>
+          </calcite-block-group>`,
+      );
+      await page.waitForChanges();
+      await page.waitForTimeout(DEBOUNCE.nextTick);
+
+      let component1Moves = 0;
+      let component2Moves = 0;
+
+      // Workaround for page.spyOnEvent() failing due to drag event payload being serialized and there being circular JSON structures from the payload elements. See: https://github.com/Esri/calcite-design-system/issues/7643
+      await page.$eval("#component1", (blockGroup: BlockGroup["el"]) => {
+        const testWindow = window as TestWindow;
+        testWindow.component1CalledTimes = 0;
+        blockGroup.addEventListener("calciteBlockGroupOrderChange", (event: CustomEvent<BlockDragDetail>) => {
+          testWindow.component1CalledTimes++;
+          testWindow.newIndex = event.detail.newIndex;
+          testWindow.oldIndex = event.detail.oldIndex;
+          testWindow.fromEl = event.detail.fromEl.id;
+          testWindow.toEl = event.detail.toEl.id;
+          testWindow.el = event.detail.dragEl.id;
+        });
+      });
+
+      // Workaround for page.spyOnEvent() failing due to drag event payload being serialized and there being circular JSON structures from the payload elements. See: https://github.com/Esri/calcite-design-system/issues/7643
+      await page.$eval("#component2", (blockGroup: BlockGroup["el"]) => {
```

Comment:
You might be able to use the new [`createEventTimePropValuesAsserter`](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/tests/utils.ts#L600C23-L600C56) for some of these assertions. 

I'll create a refactor issue to wire it up in more tests.

Reply:
Can you create an issue to tackle it in both List and BlockGroup components?